src/porepy/numerics/interface_laws/elliptic_interface_laws.py

```diff
@@ -851,3 +851,179 @@ def assemble_matrix_rhs(
+    ):
+        """Discretize the Peaceman interface law and store the discretization in the
+        edge data.
+
+        Parameters:
+            g_h: Grid of the primary domanin.
+            g_l: Grid of the secondary domain.
+            data_h: Data dictionary for the primary domain.
+            data_l: Data dictionary for the secondary domain.
+            data_edge: Data dictionary for the edge between the domains.
+
+        Implementational note: The computation of equivalent radius is highly simplified
+        and ignores discretization and anisotropy effects. For more advanced alternatives,
+        see the MRST book, https://www.cambridge.org/core/books/an-introduction-to-
+        reservoir-simulation-using-matlabgnu-octave/F48C3D8C88A3F67E4D97D4E16970F894
+        """
+        matrix_dictionary_edge: Dict[str, sps.spmatrix] = data_edge[
+            pp.DISCRETIZATION_MATRICES
+        ][self.keyword]
+        parameter_dictionary_edge: Dict = data_edge[pp.PARAMETERS][self.keyword]
+
+        parameter_dictionary_h: Dict = data_h[pp.PARAMETERS][self._primary_keyword]
+        parameter_dictionary_l: Dict = data_l[pp.PARAMETERS][self.keyword]
+        # Mortar data structure.
+        mg: pp.MortarGrid = data_edge["mortar_grid"]
+        # projection matrix
+        proj_h = mg.primary_to_mortar_avg()
+        proj_l = mg.secondary_to_mortar_avg()
+
+        r_w = parameter_dictionary_l["well_radius"]
+        skin_factor = parameter_dictionary_edge["skin_factor"]
+        # Compute equivalent radius for Peaceman well model (see above note)
+        r_e = 0.2 * np.power(g_h.cell_volumes, 1 / g_h.dim)
+        # Compute effective permeability
+        k: pp.SecondOrderTensor = parameter_dictionary_h["second_order_tensor"]
+        if g_h.dim == 2:
+            R = pp.map_geometry.project_plane_matrix(g_h.nodes)
+            k = k.copy()
+            k.values = np.tensordot(R.T, np.tensordot(R, k.values, (1, 0)), (0, 1))
+            k.values = np.delete(k.values, (2), axis=0)
+            k.values = np.delete(k.values, (2), axis=1)
+        kx = k.values[0, 0]
+        ky = k.values[1, 1]
+        ke = np.sqrt(kx * ky)
+
+        # Kh is ke * specific volume, but this is already captured by k
+        Kh = proj_h * ke * mg.cell_volumes
+        inv_WI = sps.diags((np.log(proj_h * r_e / r_w) + skin_factor) / 2 * np.pi * Kh)
+        matrix_dictionary_edge[self.well_discr_matrix_key] = -inv_WI
+
+        ## Vector source.
+        # This contribution is last term of
+        # lambda = -\int{\kappa_n [p_l - p_h +  a/2 g \cdot n]} dV,
+        # where n is the outwards normal and the integral is taken over the mortar cell.
+        # (Note: This assumes a P0 discretization of mortar fluxes).
+
+        # Ambient dimension of the problem, as specified for the higher-dimensional
+        # neighbor.
+        # IMPLEMENTATION NOTE: The default value is needed to avoid that
```

Comment:
In my opinion, this is really an argument why vector sources by default should not be discretized. Todo.